advanced/Templates/gravity.db.sql

```diff
@@ -46,44 +87,52 @@ CREATE TABLE info
 
 INSERT INTO info VALUES("version","1");
 
-CREATE VIEW vw_gravity AS SELECT a.domain
-    FROM gravity a
-    WHERE a.domain NOT IN (SELECT domain from whitelist WHERE enabled == 1);
+CREATE VIEW vw_gravity AS SELECT domain
+    FROM gravity
+    WHERE domain NOT IN (SELECT domain from vw_whitelist);
 
-CREATE VIEW vw_whitelist AS SELECT a.domain
-    FROM whitelist a
-    WHERE a.enabled == 1
-    ORDER BY a.id;
+CREATE VIEW vw_whitelist AS SELECT domain
+    FROM whitelist
+    LEFT JOIN whitelist_by_group ON whitelist_by_group.whitelist_id = whitelist.id
+    LEFT JOIN "group" ON "group".id = whitelist_by_group.group_id
+    WHERE whitelist.enabled = 1 AND ("group".enabled IS NULL OR "group".enabled = 1)
```

Comment:
Instead of checking if `"group".enabled` is null, it would be clearer to check if `whitelist_by_group.group_id` is null.